File: Lab03/Python/nprime.py

def nPierwsza(n):
    tab = []
    tab.append(2)
    licznik = 3
    k = 1

    while(k < n):
        flag = True

        i = 0

        while(i < k and tab[i] * tab[i] <= licznik and flag):
            if(licznik % tab[i] == 0):
                flag = False
            i += 1
        
        if flag:
            tab.append(licznik)
            k += 1
        licznik += 2
        
    return tab


def main():
    n = int(input("Podaj n: "))
    if n > 0:
        tablica = nPierwsza(n)

        print(f"{n} liczba pierwsza to {tablica[n-1]}")
    else:
        print("Błędne dane")
    return 0


if __name__ == "__main__":
    main()

File: Lab03/C/nprime.c
#include <stdio.h>
#include <stdbool.h>

void nPierwsza(long long n, long long *tab)
{
    tab[0] = 2;
    long long licznik = 3;
    long long k = 1; // ilosc liczb juz znalezionych

    while (k < n)
    {
        bool flag = true;

        int i = 0;
        while (i < k && (long long)tab[i] * tab[i] <= licznik && flag)
        {
            if (licznik % tab[i] == 0)
            {
                flag = false;
            }

            i++;
        }
        if (flag)
        {
            tab[k] = licznik;
            k++;
        }
        licznik += 2;
    }
}

int main()
{
    const long long maxN = 10000;
    long long tablica[maxN];
    long long n = 0;
    printf("Podaj n większe od 0: ");
    scanf("%lld", &n);
    if (n > 0)
    {
        nPierwsza(n, tablica);
        printf("%lld liczba pierwsza to: %lld\n", n, tablica[n - 1]);
    }
    else
    {
        printf("Błędne dane\n");
    }

    return 0;
}